jest.config.ts

```diff
@@ -9,6 +9,8 @@ const config: JestConfigWithTsJest = {
   moduleNameMapper: {
     '^(\\.{1,2}/.*/llhttp\\.wasm\\.js)$': '$1',
     '^(\\.{1,2}/.*)\\.js$': '$1',
+    '^@tanstack/react-query$':
+      '<rootDir>/node_modules/@tanstack/react-query/build/modern/index.cjs',
```

Comment:
For some reason, Jest resolved imports to `@tanstack/react-query` to the **ESM code** when imported from the test file, and to the **CJS code** when imported from `ra-core`. This caused the issue about `QueryClient` because the references to the context object were not equal.

This line forces the resolution to the CJS code all the time, which solves the issue.

Reply:
Isn't it a problem react-admin users may be facing, too?

Reply:
Indeed it could be, if they are using Jest with `NODE_OPTIONS=--experimental-vm-modules`. But I'm not sure it's react-admin's responsibility to document this, as this rather looks like a Jest bug to me...